### PR TITLE
Add ControlGame and Population event classes

### PR DESCRIPTION
**Pull request by Mehuge**
These will be used by the ControlGame and ControlGameScore event topics in cu-events.

ControlGame emits a ControlGame instance, which includes control point information.

ControlGameScore emits an object containing both a ControlGame instance (without control point information) and a Population instance (sourced from two different REST API endpoints)

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-core/pull/8" target="_blank">ReviewNinja

File: src/ts/classes/ControlGame.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import race from '../constants/race';
+import archetype from '../constants/archetype';
+
+class ControlGame {
+
+  arthurianScore: number;
+  controlPoints: any[];
```
Comment (codecorsair) [inline src/ts/classes/ControlGame.ts]: controlPoints should have a defined type as well.  Their format is: {id, x, y, faction, size}

File: src/ts/main.ts [hunk 2]
```diff
@@ -50,5 +52,7 @@ export {
   // classes
   Ability,
   Combatant,
-  Player
+  Player,
+  ControlGame,
+  Population
```
Comment (codecorsair) [inline src/ts/main.ts]: trailing comma is fine here, that way we don't have to modify an additional line.

File: src/ts/classes/ControlGame.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+export interface ControlPoints {
```
Comment (codecorsair) [inline src/ts/classes/ControlGame.ts]: This is a singular ControlPoint please. :smile:

File: src/ts/classes/ControlGame.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  faction: string;
```
Comment (codecorsair) [inline src/ts/classes/ControlGame.ts]: *note to self... we should send this down as the faction enum value.

File: src/ts/classes/ControlGame.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    // Game Score
+    this.arthurianScore = controlGame.arthurianScore || 0;
+    this.tuathaDeDanannScore = controlGame.tuathaDeDanannScore || 0;
+    this.vikingScore = controlGame.vikingScore || 0;
+
+    // Time remaining
+    this.timeLeft = controlGame.timeLeft || 0;
+
+    // Control Points (if included)
+    this.controlPoints = controlGame.controlPoints || null;
```
Comment (codecorsair) [inline src/ts/classes/ControlGame.ts]: this should be initialized to an empty array rather than null.

**Comment by codecorsair** [comment]
Good to merge once these 2 things are fixed.

File: src/ts/classes/ControlGame.ts (hunk 5 above)
Comment (Mehuge) [inline src/ts/classes/ControlGame.ts]: !fixed

File: src/ts/classes/ControlGame.ts (hunk 3 above)
Comment (Mehuge) [inline src/ts/classes/ControlGame.ts]: !fixed

File: src/ts/classes/ControlGame.ts (hunk 1 above)
Comment (Mehuge) [inline src/ts/classes/ControlGame.ts]: !fixed

File: src/ts/main.ts (hunk 2 above)
Comment (Mehuge) [inline src/ts/main.ts]: !fixed

**Pull request merged**